presto-parser/src/main/java/com/facebook/presto/sql/SqlFormatter.java

```diff
@@ -860,7 +864,11 @@ private static String formatName(QualifiedName name)
 
         private String formatColumnDefinition(ColumnDefinition column)
         {
-            return formatExpression(column.getName(), parameters) + " " + column.getType() +
+            String columnName = formatExpression(column.getName(), parameters);
+            columnName = RESERVED_IDENTIFIERS.contains(columnName.toUpperCase(ENGLISH)) ?
```

Comment:
This only fixes reserved words, but not other characters that require quoting, such as spaces. Also, any other place that formats identifiers or names will have the same problem.

For now, let's continue to ignore the semantic difference between delimited and non-delimited identifiers, and change this to use the `formatName` which is used for everything else:

```java
formatName(column.getName().getValue())
```

The existing `NAME_PATTERN` logic in `Identifier` probably needs to handle reserved words, or needs to go away entirely, but let's leave that for another day.

Reply:
If a column name requires quoting, then the `Identifier` must be delimited, and thus `formatExpression` produces correct formatting by adding quotes for delimited identifier. You can see the test for space below.

`formatName` does not fix the described issue. Column with reserved words is still formatted as unquoted, leading to a syntactically invalid create table statement.

Reply:
Yes, but my point is that anything using formatName() is broken, so that’s the correct place to make the change. The fix is not specific to column definitions (and thus column definition formatting should not be aware of this issue).